File: Leetcode3-LinkedLists&Arrays/LongestCommonPrefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = strs[0]

        for stri in strs:
            temp = ""
            i = 0
            while i < min(len(prefix),len(stri)) and prefix[i] == stri[i]:
                temp += stri[i]
                i+=1
            prefix = temp


        return prefix

File: Leetcode3-LinkedLists&Arrays/LongestCommonPrefix/solution.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string prefix = strs[0];

        for (int i = 1; i < strs.size(); i++){
            string str = strs[i];
            string temp = "";

            int j = 0;

            while (j < min(str.length(), prefix.length()) && str[j] == prefix[j]){
                temp+=str[j];
                j++;
            }
            prefix = temp;
        }

        return prefix;
    }
};